src/ens.sol

```diff
@@ -0,0 +1,50 @@
+interface IReverseRegistrar {
+    function claim(address owner) external returns (bytes32 node);
+    function claimWithResolver(address owner, address resolver) external returns (bytes32 node);
+    function setName(string calldata name) external returns (bytes32 node);
+    function node(address) external view returns (bytes32 node);
+}
+
+interface INameResolver {
+    function name(bytes32 node) external view returns (string memory);
+}
+
+interface IENS {
+    function setSubnodeRecord(bytes32 node, bytes32 label, address owner, address resolver, uint64 ttl) external;
+    function resolver(bytes32 node) external view returns (address);
+    function owner(bytes32 node) external view returns (address);
+    function setResolver(bytes32 node, address resolver) external;
+}
+
+interface IAddrResolver {
+    function addr(bytes32 node) external view returns (address);
+}
```

Comment:
Where are all these imported / copied from?

Reply:
Directly from the contracts, preferred copying the interfaces(very few lines) vs adding the huge ens monorepo as a dependency